packages/vite/src/node/plugins/reporter.ts

```diff
@@ -91,6 +91,10 @@ export function buildReporterPlugin(config: ResolvedConfig): Plugin {
       startTime = Date.now()
     },
 
+    buildStart() {
+      transformedCount = 0
+    },
```

Comment:
Added one unrelated change here while I'm at it 😬 This makes it so that modules transformed after `--watch` is reset so we get the real count.